src/init/features/hosting/github.ts

```diff
@@ -371,14 +375,16 @@ async function uploadSecretToGitHub(
   keyId: string,
   secretName: string
 ): Promise<{ status: any }> {
-  return await api.request("PUT", `/repos/${repo}/actions/secrets/${secretName}`, {
-    origin: api.githubApiOrigin,
-    headers: { Authorization: `token ${ghAccessToken}`, "User-Agent": "Firebase CLI" },
-    data: {
-      ["encrypted_value"]: encryptedServiceAccountJSON,
-      ["key_id"]: keyId,
-    },
-  });
+  const data = {
+    ["encrypted_value"]: encryptedServiceAccountJSON,
+    ["key_id"]: keyId,
+  };
+  const headers = { Authorization: `token ${ghAccessToken}`, "User-Agent": "Firebase CLI" };
+  return await githubApiClient.put<any, { status: any }>(
+    `/repos/${repo}/actions/secrets/${secretName}`,
+    data,
+    { headers }
+  );
```

Comment:
This errors out for me when I try to run `firebase init` in this branch. The error is:

```
FetchError: invalid json response body at https://api.github.com/repos/jhuleatt/hosting-deploy-sample/actions/secrets/FIREBASE_SERVICE_ACCOUNT_WORDS_TO_HONK reason: Unexpected end of JSON input
    at /Users/jhuleatt/git/firebase-tools/node_modules/node-fetch/lib/index.js:272:32
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

I've confirmed that this doesn't happen in the current latest `firebase-tools` version, `8.17.0`.

Reply:
Turns out that 204 responses with no content don't parse to JSON very well! I added logic to prevent this issue.